plenum/server/node.py

```diff
@@ -2130,6 +2125,161 @@ def _is_there_pool_ledger(self):
         # TODO isinstance is not OK
         return isinstance(self.poolManager, TxnPoolManager)
 
+    def process_message_req(self, msg: MessageReq, frm):
+        # Assumes a shared memory architecture. In case of multiprocessing,
+        # RPC architecture, use deques to communicate the message and node will
+        # maintain a unique internal message id to correlate responses.
+        resp = None
+        msg_type = msg.msg_type
+        if msg_type == LEDGER_STATUS:
+            params = msg.params
+            ledger_id = params.get(f.LEDGER_ID.nm)
+            if self.valid_requested_msg(msg_type, ledger_id=ledger_id):
+                resp = self.getLedgerStatus(ledger_id)
+            else:
+                self.discard(msg, 'cannot serve request',
+                             logMethod=logger.debug)
+                return
+        elif msg_type == CONSISTENCY_PROOF:
+            params = msg.params
+            ledger_id = params.get(f.LEDGER_ID.nm)
+            seq_no_start = params.get(f.SEQ_NO_START.nm)
+            seq_no_end = params.get(f.SEQ_NO_END.nm)
+            if self.valid_requested_msg(msg_type, ledger_id=ledger_id,
+                                        seq_no_start=seq_no_start,
+                                        seq_no_end=seq_no_end):
+                resp = self.ledgerManager._buildConsistencyProof(ledger_id,
+                                                                 seq_no_start,
+                                                                 seq_no_end)
+            else:
+                self.discard(msg, 'cannot serve request',
+                             logMethod=logger.debug)
+                return
+
+        elif msg_type == PREPREPARE:
+            params = msg.params
+            inst_id = params.get(f.INST_ID.nm)
+            view_no = params.get(f.VIEW_NO.nm)
+            pp_seq_no = params.get(f.PP_SEQ_NO.nm)
+            if self.valid_requested_msg(msg_type, inst_id=inst_id,
+                                        view_no=view_no, pp_seq_no=pp_seq_no):
+                resp = self.replicas[inst_id].getPrePrepare(view_no, pp_seq_no)
```

Comment:
Yes, in that case None will be returned which is handled. If a node is not able to get required data through these messages, then it should start catchup